src/app/sitemap.ts

```diff
@@ -0,0 +1,36 @@
+import type { MetadataRoute } from "next";
+
+const isProd = process.env.NODE_ENV === "production";
+const BASE_URL = isProd ? "https://papermc.io" : "http://localhost:3000";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  const now = new Date();
+
+  const staticRoutes: MetadataRoute.Sitemap = [
+    "/community/guidelines",
+    "/downloads",
+    "/downloads/folia",
+    "/downloads/velocity",
+    "/",
+    "/software/folia",
+    "/javadocs",
+    "/community",
+    "/downloads/waterfall",
+    "/software/paper",
+    "/team",
+    "/sponsors",
+    "/software/waterfall",
+    "/software/velocity",
+    "/downloads/paper",
+    "/contribute",
+  ].map((route) => ({
+    url: new URL(route, BASE_URL).toString(),
+    lastModified: now,
```

Comment:
This is ran at build time, right? So last modified gets updated only on a new build, not on every request?

Reply:
It should be, yes. 

Documentation: https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap

- The `sitemap.ts` file is a special Route Handler that's cached by default (unless it uses dynamic APIs or config).
- It's generated during the build process as part of static metadata generation.
- The `lastModified` field is set at build time (e.g., `new Date()` evaluates to the build timestamp), and the resulting XML is static until the next build.